src/worker_thread.c

```diff
@@ -100,39 +106,14 @@ WORKER_THREAD_HANDLE worker_thread_create(WORKER_FUNC worker_func, void* worker_
             {
                 /* Codes_SRS_WORKER_THREAD_01_022: [ worker_thread_create shall perform the following actions in order: ]*/
 
-                /* Codes_SRS_WORKER_THREAD_01_006: [ worker_thread_create shall create an auto reset unnamed event used for signaling the thread to shutdown by calling CreateEvent. ]*/
-                worker_thread->worker_thread_shutdown_event = CreateEvent(NULL, FALSE, FALSE, NULL);
-                if (worker_thread->worker_thread_shutdown_event == NULL)
-                {
-                    /* Codes_SRS_WORKER_THREAD_01_008: [ If any error occurs, worker_thread_create shall fail and return NULL. ]*/
-                    LogError("Cannot create shutdown event");
-                }
-                else
-                {
-                    /* Codes_SRS_WORKER_THREAD_01_007: [ worker_thread_create shall create an auto reset unnamed event used for signaling that a new execution of the worker function should be done by calling CreateEvent. ]*/
-                    worker_thread->execute_worker_event = CreateEvent(NULL, FALSE, FALSE, NULL);
-                    if (worker_thread->execute_worker_event == NULL)
-                    {
-                        /* Codes_SRS_WORKER_THREAD_01_008: [ If any error occurs, worker_thread_create shall fail and return NULL. ]*/
-                        LogError("Cannot create process event");
-                    }
-                    else
-                    {
-                        goto all_ok;
-
-                        //(void)CloseHandle(worker_thread->execute_worker_event);
-                    }
-
-                    (void)CloseHandle(worker_thread->worker_thread_shutdown_event);
-                }
-
-                sm_destroy(worker_thread->sm);
+                /* Codes_SRS_WORKER_THREAD_01_006: [ worker_thread_create shall initialize the internal objects. ]*/
+                (void)interlocked_exchange(&worker_thread->state, 0);
```

Comment:
>0 [](http://example.com/codeflow?start=66&length=1)

what's the meaning of this "0" here? `WORKER_THREAD_STATE` enum doesn't have it listed explicitly.